2.0.0 changes

cs-abhishek-shukla opened the pull request:
- release notes WIP
- usage updated
- Updated setup
- contents wip

cs-abhishek-shukla committed:
2.0.0 changes
- release notes WIP
- usage updated
- Updated setup
- contents wip

cs-suyog-jinturkar commented on README.md:
Create a new page for threat hunt rules and add all removed in it and link it here

cs-suyog-jinturkar replied:
Removed info is useful info the user can be retained.

cs-suyog-jinturkar replied:
@cs-abhishek-shukla  this is still pending

cs-abhishek-shukla replied:
The removed info is now a part of a new file [Overview of Outbreak Response Framework](https://github.com/cs-abhishek-shukla/solution-pack-outbreak-response-framework/blob/release/2.0.0/docs/background-information.md)

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -0,0 +1,16 @@
+# What's New
+
+- Following new pages have been added:
```
These notes are same as config wizard. It can't be same here in SP. You can create heading as configuration wizard enhancements and then describe details wrt user experience and not as a technicality of the wizard, i.e. what is removed or added.

cs-suyog-jinturkar replied:
Pluggable hunt framework details are still missing from RN and usage section

cs-abhishek-shukla committed:
2.0.0 changes
- Updated release notes as per information in the doc
- Updated contents as per the same doc

cs-abhishek-shukla committed:
2.0.0 changes
- Created and added content to Pluggable threat hunting
- Create and added content to Triggering and monitoring investigation using PB button and pb exec wizard
- Added necessary SS
- Create and added content to Overview of Outbreak Response Framework (rewritten content removed from Readme.md)
- Removed Troubleshooting section and flowchart from Usage

cs-suyog-jinturkar commented on docs/res/dashboard-outbreak-response-overview.png:
resolution should be 1920x1080. please correct it. Please scroll down and add 2nd half of the DB SS as well

cs-suyog-jinturkar commented on docs/res/pb-exec-wizard.gif:
gif is okey but Its not giving me a clear picture of the screens. Can you please also add static SS?

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -83,9 +83,13 @@ After installation of the **Outbreak Response Framework** solution pack, run the
+
+    Click the button **Ingest Now** to install the outbreak response solution packs of the severity selected on the previous screen.
+
+    Click **Finish** to complete the configuration process.
```
You will need to elaborate more on what things happen in the background as part of the finish page and overall configuration wizard like it creates 1,2,3 schedules for xyz things, stores hunt tools details and its param in key-store, etc.

When can user expect the installation of OBR SPs starts, same for invitation PBs, etc.

Please collaborate with Sujit for completing this section.

cs-suyog-jinturkar commented on docs/trigger-and-monitor-pb-with-peb.md:
[trigger-and-monitor-pb-with-peb.md] --> what is "peb" means here in the filename?

cs-suyog-jinturkar replied:
i think, playbook execution button, correct?

cs-dev-chaudhary commented on README.md:
```diff
@@ -1,118 +1,42 @@
 # Release Information 
 
-- **Version**: 1.0.0![](./docs/res/icon-preview.svg)
+- **Version**: 2.0.0![](./docs/res/icon-preview.svg)
```
Preview needs to be removed.

cs-suyog-jinturkar replied:
please do not remove preview tag

cs-dev-chaudhary commented on README.md:
Screenshot needs to be changed, last column is blank, we can edit the SVT and take out the new SS.

cs-dev-chaudhary commented on docs/contents.md:
```diff
@@ -7,14 +7,21 @@ The **Outbreak Response Framework** solution pack contains the following resourc
+| Threat Hunt Rule Type![Renamed](./res/icon-renamed.svg) | (Previously *Rule Type*) List of Outbreak Alert Threat Hunt Rule item values (Yara, Sigma, Fortinet Fabric) |
+| Threat Hunt Tools                                       | List of Threat Hunt Integration item value which will used for Threat Hunting                               |
+| Outbreak Alert Status![Renamed](./res/icon-renamed.svg) | (Previously *Record Status*) List of Outbreak Alert status item value(Active, Resolved)                     |
+
```
IncidentType is missing.  Its also there in content.

cs-abhishek-shukla replied:
Please provide a description as well

cs-dev-chaudhary commented on docs/contents.md:
```diff
@@ -25,66 +32,60 @@ The **Outbreak Response Framework** solution pack contains the following resourc
+| Fortinet FortiAnalyzer                                         | FortiAnalyzer is the NOC-SOC security analysis tool built with operations perspective. With action-oriented views and deep drill-down capabilities, FortiAnalyzer not only gives organizations critical insight into threats, but also accurately scopes risk across the attack surface, pinpointing where immediate response is required.   ...
+| Fortinet FortiSIEM                                             | FortiSIEM provides integrations that allow you to query and make changes to the CMDB, query events, and send incident notifications. Provide actions like get incidents, comment incident, cleared incident, get device details, get monitored organizations, report related actions and get all associated events for an incident from Fort ...
+| IBM QRadar                                                     | IBM QRadar is an enterprise security information and event management (SIEM) product. Fortinet FortiSOSR connector for IBM QRadar allows user to invoke QRadar API, perform Ariel Queries and operations like Get Offense,related events,update and close offenses.                                                                          ...
+| Fortinet FortiGate                                             | Fortinet FortiGate enterprise firewall provide high performance, consolidated advanced security and granular visibility for broad protection across the entire digital attack surface.                                                                                                                                                       ...
```
Removed.

cs-dev-chaudhary commented on docs/contents.md:
```diff
@@ -25,66 +32,60 @@ The **Outbreak Response Framework** solution pack contains the following resourc
+| NIST National Vulnerability Database                           | The NIST National Vulnerability Database (NVD) is the U.S. government repository of standards based vulnerability management data represented using the Security Content Automation Protocol (SCAP). This data enables automation of vulnerability management, security measurement, and compliance. The NVD includes databases of security  ...
+| Splunk                                                         | Splunk connector allows users to invoke search, fetch events to related search, invoke alert actions, update notables, sync splunk users to FortiSOAR etc.                                                                                                                                                                                   ...
+| Azure Log Analytics![Removed](./res/icon-removed.svg)          | Log Analytics is a tool in the Azure portal that's used to edit and run log queries against data in the Azure Monitor Logs store. This connector facilitates the automated operations related to query and saved searches.                                                                                                                   ...
+| Fortinet FortiGuard Outbreak![Removed](./res/icon-removed.svg) | Fortinet FortiGuard Outbreak connector receives communication with *FortiGuard Outbreak Alerts* regarding an outbreak and its details. These alerts help understand the technical details of the attack and how organizations can protect themselves from the attack and others like it.                                                     ...
```
Exists.

cs-dev-chaudhary commented on docs/contents.md:
```diff
@@ -25,66 +32,60 @@ The **Outbreak Response Framework** solution pack contains the following resourc
 ## Playbook Collection
 
 | 10 - SP - Outbreak Response Framework |
 |:-------------------------------------:|
```
05 - Hunt collection is not listed. Please mention

cs-dev-chaudhary commented on docs/pluggable-threat-hunting.md:
```diff
@@ -0,0 +1,54 @@
+
+# Next Steps
+
+| [Installation](./setup.md#installation) | [Configuration](./setup.md#configuration) | [Usage](./usage.md) |
```
SS for wizard and installation needs to be changed.

cs-dev-chaudhary commented on docs/setup.md:
```diff
@@ -24,26 +24,19 @@ For optimal performance of the **Outbreak Response Framework** solution pack, in
 ## Setup Outbreak Response Framework on FortiSOAR
 
-After installation of the **Outbreak Response Framework** solution pack, run the configuration wizard to ready your FortiSOAR environment to investigate Outbreak Alerts. This wizard helps you select the configured "Threat Detection Integrations" and "Ticketing/ITSM Integrations" on FortiSOAR.
+After installation of the **Outbreak Response Framework** solution pack, run the configuration wizard to ready your FortiSOAR environment to investigate Outbreak Alerts. This wizard helps you select the configured **Threat Detection Integrations** on FortiSOAR.
```
This wizard helps you select the configured **Threat Detection Integrations** on FortiSOAR. ---> This wizard assists you in selecting and configuring the **Threat Detection Integrations** on FortiSOAR.

cs-dev-chaudhary commented on docs/setup.md:
```diff
@@ -55,20 +48,50 @@ After installation of the **Outbreak Response Framework** solution pack, run the
+
+7. **Installation & Notification**
+
+    ![Installation and notification page](./res/config-wizard-05.png)
```
Please change the SS to latest.

cs-dev-chaudhary commented on docs/setup.md:
```diff
@@ -55,20 +48,50 @@ After installation of the **Outbreak Response Framework** solution pack, run the
+    ![Installation and notification page](./res/config-wizard-05.png)
 
-    The hunt activities require searching for adversaries and their tactics, within an environment, against existing information available in the Threat Intel Platform's database (TIP). The Threat Detection Integration sources help run the threat hunt activities and are an important part of the *Outbreak Response Framework*.
+    - **Automatically install outbreak response solution packs**: Select this option to automatically install outbreak response solution packs once an outbreak is reported.
```
This part has been removed lately. Please refer to latest build

cs-dev-chaudhary commented on docs/setup.md:
```diff
@@ -55,20 +48,50 @@ After installation of the **Outbreak Response Framework** solution pack, run the
+    ![All set](./res/config-wizard-06.png)
 
 ## Configure the Threat Detection Integration Parameters
 
```
This needs to be revised entirely due to changes in the investigate playbook and the way we handle input parameters for all Threat Integrations.

cs-dev-chaudhary commented on docs/usage.md:
```diff
@@ -43,15 +46,13 @@ The following is an ideal flow to use the **Outbreak Response Framework** soluti
-8. **Ticketing/ITSM**: If any vulnerability is found using the IOC Threat Hunt or Signature Based Threat Hunt automatically Alert is created for that vulnerability in FortiSOAR and as part of your response or threat management strategy automatically creates tickets using Ticketing/ITSM Integrations (Jira/ServiceNow) for Outbreak Alert(vulnerability). This step is performed only if you have added  ...
-
-9. **Remediation**: We can take remediation using two ways
+8. **Remediation**: We can take remediation using two ways
```
This part have been removed now. We dnt use fortigate anymore to block the IOCs.

cs-dev-chaudhary commented on docs/usage.md:
```diff
@@ -43,15 +46,13 @@ The following is an ideal flow to use the **Outbreak Response Framework** soluti
     2. **Manually**: FortiSOAR automatically creates a FortiSOAR task to block all the indicators using the FortiGate Integration.
 
-10. **Mitigation**: For every Outbreak Alert have associated mitigation. FortiSOAR provides the mitigation recommendations using public sources, like patch available, etc.
+9. **Mitigation**: For every Outbreak Alert have associated mitigation. FortiSOAR provides the mitigation recommendations using public sources, like patch available, etc.
```
We are creating incidents now instead of alerts.

cs-dev-chaudhary commented on release_notes.md:
```diff
@@ -0,0 +1,63 @@
+- Removed dependency on the following connectors as IT/SM tools are no longer supported:
+    - Azure Log Analytics
+    - Fortinet FortiGate
+    - Jira
```
We are creating Incidents now not alerts. Also Fortigate remediation part is also removed. Now user can act on incident using incident response PBs.

cs-abhishek-shukla committed:
2.0.0 changes
- release notes revamp

cs-dev-chaudhary commented on release_notes.md:
```diff
@@ -0,0 +1,63 @@
+    - `outbreak-threat-hunt-tools`
+    - `outbreak-auto-install-time-frame`
+    - `outbreak-threat-hunt-tools-params`
+
```
Schedules needs to be mentioned both in content and release notes.

cs-abhishek-shukla committed:
2.0.0 changes
- release notes changes at RT

cs-abhishek-shukla committed:
2.0.0 changes
- release notes changes at RT

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -0,0 +1,94 @@
+
+### Outbreak Management - A New Look!
+
+Get ready to experience a revamped Outbreak Management interface designed to streamline your workflow and enhance your visibility. When you click *Outbreak Management* in the navigation menu, you’re greeted with an intuitive layout featuring these powerful tabs:
```
Please see below suggested text;

Get ready to experience a revamped Outbreak Management, designed to streamline your investigation workflow and enhance your visibility of your environment to respond and remediate these Outbreak Alerts effectively and efficiently. 
When you click *Outbreak Management* in the navigation menu, you’re greeted with an intuitive layout featuring these powerful tabs:

cs-abhishek-shukla replied:
Done

cs-suyog-jinturkar commented on README.md:
See if we can add below information as well copied from FGD website;

FortiGuard Outbreak Alerts provides key information about on-going cybersecurity attack with significant ramifications affecting numerous companies, organizations and industries. These Outbreak Alerts will help you understand what happened, the technical details of the attack and how organizations can protect themselves from the attack and others like it.

The Alert will include:
- Details of the attack including timeline, technology affected, and where applicable patches/ mitigation recommendations can be found
- Recommended Fortinet products that would break the attack sequence, and threat hunting tools to help you determine if you were affected
- Additional related research from FortiGuard Labs

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -0,0 +1,94 @@
+
+The wizard has received several key upgrades, making your configuration process smoother and more efficient:
+
+- Each integration now has its dedicated configuration tab for a more organized setup experience.
```
and avoid missing key configuration settings.

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -0,0 +1,94 @@
+  
+- Take control of your security operations by automating investigation frequencies and tailoring your investigation windows to fit your needs.
+  
+- Never miss an update! Opt for automatic installation and get near-instant notifications when outbreak-specific solution packs are available.
```
when outbreak-specific solution packs are available --> when outbreak-specific solution packs are installed.

cs-abhishek-shukla committed:
2.0.0 changes
- release notes changes at RT

cs-abhishek-shukla committed:
2.0.0 changes
- release notes changes at RT
- all feedback minus JSON

cs-suyog-jinturkar reviewed (approved):
Almost of the commented are addressed. Remaining one will be fixed in upcoming PR.

cs-suyog-jinturkar commented on README.md:
Please add release note link here

cs-suyog-jinturkar commented on release_notes.md:
```diff
@@ -0,0 +1,115 @@
+
+### Outbreak Management - A New Look!
+
+Get ready to experience a revamped Outbreak Management, designed to streamline your investigation workflow and enhance your visibility of your environment to respond and remediate these Outbreak Alerts effectively and efficiently.
```
We will need to mention that this SP is compatible only FSR 7.6 onwards.

cs-suyog-jinturkar commented on docs/setup.md:
```diff
@@ -13,9 +13,9 @@ The **Outbreak Response Framework** solution pack depends on the following solut
+| Threat Intel Management  | Solution Pack | v1.2.2  and later | Required to ingest threat feeds        |
+| Vulnerability Management | Solution Pack | v2.1.0  and later | Required to ingest CVEs for KEVs       |
 
 # Configuration
```
We will mention that this SP is tested with FortiCloud FAZ 7.4.3 and on-prem 7.4.2

cs-abhishek-shukla replied:
not putting this info

cs-dev-chaudhary commented on docs/usage.md:
```diff
@@ -77,55 +70,39 @@ Before performing the steps outlined in this section, we recommend setting the g
 
 | [Installation](./setup.md#installation) | [Configuration](./setup.md#configuration) | [Contents](./contents.md) |
-|-----------------------------------------|-------------------------------------------|---------------------------|
+|-----------------------------------------|-------------------------------------------|---------------------------|
```
Need to update known issues after upgrade:
1. Need to add "Deactivated" status to outbreak alert status.
2. Edit the step "Update Last Pull Time" field " status to "tracking" in "> Investigate Outbreak Alert" PB.
3. In Outbreak Alert Time Frame Analysis last step "Update Outbreak Alert Status" user needs to select status as "deactivated"
4.
